tests/integration/cmd_dev_recoverability_test.go

```diff
@@ -0,0 +1,117 @@
+//go:build linux || darwin || dragonfly || solaris || openbsd || netbsd || freebsd
+// +build linux darwin dragonfly solaris openbsd netbsd freebsd
+
+package integration
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/redhat-developer/odo/tests/helper"
+)
+
+var _ = Describe("Recoverability Test", func() {
+	var commonVar helper.CommonVar
+	var _ = BeforeEach(func() {
+		commonVar = helper.CommonBeforeEach(helper.SetupClusterTrue)
+		// helper.Chdir(commonVar.Context)
+		// helper.CopyExample(filepath.Join("source", "latency"), commonVar.Context)
+	})
+	var _ = AfterEach(func() {
+		helper.CommonAfterEach(commonVar)
+	})
+
+	checkIfDevEnvIsUp := func(url, assertString string) {
+		Eventually(func() string {
+			resp, err := http.Get(fmt.Sprintf("http://%s", url))
+			Expect(err).ToNot(HaveOccurred())
+			defer resp.Body.Close()
+
+			body, _ := io.ReadAll(resp.Body)
+			return string(body)
+		}, 120*time.Second, 15*time.Second).Should(Equal(assertString))
+	}
+
+	Context("starting with empty Directory", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			helper.Chdir(commonVar.Context)
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(BeEmpty())
+		})
+
+		It("should verify developer workflow from empty Directory", func() {
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				helper.ExpectString(ctx, "Select language")
+				helper.SendLine(ctx, "javascript")
+
+				helper.ExpectString(ctx, "Select project type")
+				helper.SendLine(ctx, "Node.js\n")
+
+				helper.ExpectString(ctx, "Which starter project do you want to use")
+				helper.SendLine(ctx, "nodejs-starter\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("server.js"))
+
+			// "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
```

Comment:
You will need to Kill() or Stop() the session before the end of the test, or the odo process will run infinitely

Reply:
Thank you @feloy . Added.